diff --git a/dist/source/as-promise/index.js b/dist/source/as-promise/index.js
index b4026d344ffd2ae2924613529167e6a80826cc95..91ff2d125a5c581a1d13610ab923076068230e4f 100644
--- a/dist/source/as-promise/index.js
+++ b/dist/source/as-promise/index.js
@@ -30,9 +30,14 @@ function asPromise(normalizedOptions) {
     let globalRequest;
     let globalResponse;
     const emitter = new events_1.EventEmitter();
-    const promise = new PCancelable((resolve, reject, onCancel) => {
+    const promise = new PCancelable((_resolve, _reject, _onCancel) => {
+        const resolved = false;
+        const resolve = (value) => {resolved = true; _resolve(value)}
+        const reject = (value) => {resolved = true; _reject(value)}
+        const onCancel = (value) => {if(!resolved) {_onCancel(value)}}
         const makeRequest = (retryCount) => {
             const request = new core_1.default(undefined, normalizedOptions);
+            
             request.retryCount = retryCount;
             request._noPipe = true;
             onCancel(() => request.destroy());
