diff --git a/dist/launcher.js b/dist/launcher.js
index f4fa8a656c9ac9588fa4c0afce54a91acccc78db..cff2c926fb240f810c7e780e9b779250faa8051b 100644
--- a/dist/launcher.js
+++ b/dist/launcher.js
@@ -131,7 +131,9 @@ export default class VSCodeServiceLauncher extends ChromedriverServiceLauncher {
         try {
             const chromedriverVersion = await this._fetchChromedriverVersion(version);
             log.info(`Download Chromedriver (v${chromedriverVersion})`);
-            await downloadBundle(format(CHROMEDRIVER_DOWNLOAD_PATH, chromedriverVersion, validatePlatform()), this._cachePath, { extract: true, strip: 1 });
+            const rawDownloadPath = format(CHROMEDRIVER_DOWNLOAD_PATH, chromedriverVersion, validatePlatform());
+            const downloadPath = parseInt(chromedriverVersion.split(".")[0]) > 106 ? rawDownloadPath.replace("64_m1", "arm64") : rawDownloadPath;
+            await downloadBundle(downloadPath, this._cachePath, { extract: true, strip: 1 });
             const ext = os.platform().startsWith('win') ? '.exe' : '';
             const chromedriverPath = path.join(this._cachePath, `chromedriver-${chromedriverVersion}${ext}`);
             await fs.rename(path.join(this._cachePath, `chromedriver${ext}`), chromedriverPath);
diff --git a/dist/service.js b/dist/service.js
index cec25dddb97ec6524e0411cc59387a6242a67a17..d05213c8808bb552a40a8bab2e9c446e4d0afb24 100644
--- a/dist/service.js
+++ b/dist/service.js
@@ -56,7 +56,7 @@ export default class VSCodeWorkerService {
             return;
         }
         const customArgs = { ...VSCODE_APPLICATION_ARGS };
-        const storagePath = await tmp.dir();
+        const storagePath = this._vscodeOptions.storagePath ? {path: this._vscodeOptions.storagePath} : await tmp.dir();
         const userSettingsPath = path.join(storagePath.path, 'settings', 'User');
         const userSettings = {
             ...DEFAULT_VSCODE_SETTINGS,
